tests/integration/helpers/helpers.py

```diff
@@ -202,15 +204,15 @@ def wait_for_relation_joined_between(
     try:
         for attempt in Retrying(stop=stop_after_delay(3 * 60), wait=wait_fixed(3)):
             with attempt:
-                if new_relation_joined(ops_test, endpoint_one, endpoint_two):
-                    break
+                if not new_relation_joined(ops_test, endpoint_one, endpoint_two):
+                    raise Exception("Relation not created")
```

Comment:
You can just assert `new_relation_joined` instead.

Reply:
addressed in 79f0de8362d1cb5f3d00125a778b4c39e1d5e91d